spring-boot-project/spring-boot-tools/spring-boot-configuration-processor/src/main/java/org/springframework/boot/configurationprocessor/TypeUtils.java

```diff
@@ -142,8 +142,7 @@ public String getJavaDoc(Element element) {
 		String javadoc = (element != null
 				? this.env.getElementUtils().getDocComment(element) : null);
 		if (javadoc != null) {
-			javadoc = javadoc.replaceAll("\\n", "");
-			javadoc = javadoc.trim();
+			javadoc = javadoc.replaceAll("[\r\n]+", "").trim();
```

Comment:
Can we do two passes instead (`\r\n` then `\n`)? Technically speaking that would replace `\r`

Reply:
@snicoll I thought removing "\r" in Javadoc, especially as description for application properties is safe. Am I missing something?